internal/appsec/appsec.go

```diff
@@ -199,6 +199,10 @@ func (a *appsec) stop() {
 		a.wafHandle.Close()
 		a.wafHandle = nil
 	}
+
+	// Reset rules edits received from the remote configuration
+	a.cfg.RulesManager, _ = config.NewRulesManager(nil)
```

Comment:
IIRC, Christophe has added start and stop system tests which you can probably use to further validate this.